webapp/views.py

```diff
@@ -453,21 +454,65 @@ def advantage_view():
 
             raise http_error
 
+        listings_response = advantage.get_marketplace_product_listings(
+            "canonical-ua"
+        )
+
+        product_listings = listings_response.get("productListings")
+
+        monthly_products = {}
+        yearly_products = {}
+        listings_to_products = {}
+
+        for listing in product_listings:
+            listings_to_products[listing["id"]] = listing["productID"]
+
+            if listing["period"] == "yearly":
+                yearly_products[listing["productID"]] = {
+                    "id": listing["id"],
+                    "metric": listing["allowanceMetric"],
+                    "price": listing["price"],
+                    "quantity": 0,
+                }
+
+            monthly_products[listing["productID"]] = {
+                "id": listing["id"],
+                "metric": listing["allowanceMetric"],
+                "price": listing["price"],
+                "quantity": 0,
+            }
+
         for account in accounts:
             account["contracts"] = advantage.get_account_contracts(account)
 
-            monthly_subscriptions = (
+            all_subscriptions = (
                 advantage.get_account_subscriptions_for_marketplace(
                     account_id=account["id"],
                     marketplace="canonical-ua",
-                    filters={
-                        "status": "active",
-                        "period": "monthly",
-                    },
+                    filters={"status": "active"},
                 )
             )
 
-            for subscription in monthly_subscriptions.get("subscriptions", []):
+            monthly_subscriptions = []
+            yearly_subscriptions = []
+
+            for subscription in all_subscriptions.get("subscriptions", []):
+                period = subscription["subscription"]["period"]
+                previous_purchase_ids[period] = subscription["lastPurchaseID"]
+
+                if subscription["subscription"]["period"] == "yearly":
+                    yearly_subscriptions.append(subscription)
+
+                monthly_subscriptions.append(subscription)
+
+            for subscription in monthly_subscriptions:
+                for purchase in subscription["purchasedProductListings"]:
```

Comment:
please name this variable to something like `purchasedProductListing` or `ppl` to be more accurate.